test/quantization/fx/test_model_report_fx.py

```diff
@@ -1715,6 +1715,77 @@ def test_get_modules_and_features(self):
             returned_plottable_feats = mod_rep_visualizer.get_all_unique_feature_names()
             self.assertEqual(returned_plottable_feats, plottable_set)
 
+    @skipIfNoFBGEMM
+    def test_generate_tables(self):
+        """
+        Tests the generate_table_view()
+        ModelReportVisualizer
+
+        Checks whether the generated dict has proper information
+            Visual check that the tables look correct performed during testing
+        """
+        with override_quantized_engine('fbgemm'):
+            # set the backend for this test
+            torch.backends.quantized.engine = "fbgemm"
+            # test with multiple detectors
+            detector_set = set()
+            detector_set.add(OutlierDetector(reference_percentile=0.95))
+            detector_set.add(InputWeightEqualizationDetector(0.5))
+
+            model = TestFxModelReportClass.TwoThreeOps()
+
+            # get tst model and callibrate
+            prepared_for_callibrate_model, mod_report = _get_prepped_for_calibration_model_helper(
+                model, detector_set, model.get_example_inputs()[0]
+            )
+
+            mod_rep_visualizer: ModelReportVisualizer = self._callibrate_and_generate_visualizer(
+                model, prepared_for_callibrate_model, mod_report
+            )
+
+            table_dict = mod_rep_visualizer.generate_filtered_tables()
+
+            # test primarily the dict since it has same info as str
+            tensor_table = table_dict[ModelReportVisualizer.TABLE_TENSOR_KEY]
```

Comment:
user still need to know the structure of the dictionary in this case, should we make this a dataclass object or something so that user can call some methods to get it?

Reply:
i feel like for now, the dictionary has relatively few keys, so they should be able to just easily look at the keys (and I don't expect it to grow anymore).

However, I think revisiting these and making them data classes is an interesting idea worth revisiting later.